pkgs/top-level/darwin-packages.nix

```diff
@@ -223,24 +223,25 @@ impure-cmds // appleSourcePackages // chooseLibs // {
   discrete-scroll = callPackage ../os-specific/darwin/discrete-scroll { };
 
   # See doc/builders/special/darwin-builder.section.md
-  linux-builder = lib.makeOverridable ({ modules }:
+  linux-builder = lib.makeOverridable ({ modules, guestSystem, hostPackages }:
```

Comment:
I think the interface makes it clearer for end users who want to run `x86_64-linux` builders on `aarch64-darwin`, maybe we could create new attributes like `x86_64-linux-builder` and `aarch64-linux-builder` possibly?

Reply:
There is no other way to configure system afaik, so at least this one is necessary to support this.

Removed hostPkgs but adjusted the settings so users can override it without mkForce, which is think would be bad ux.

Reply:
> override it without mkForce

:+1: 

> no other way to configure system

Something _should_ be possible. Maybe it gets complicated because of the `nixpkgs.system` -> `nixpkgs.hostPlatform` migration? If so, it might be time to move that migration forward by actually deprecating.